luatest/server.lua

```diff
@@ -168,11 +295,28 @@ function Server:restart(params)
         self[param] = value
     end
 
-    self:start()
+    self:start(opts)
     log.debug('Restarted server PID: ' .. self.process.pid)
 end
 
---- Stop server process.
+-- Wait until the given condition is `true` (anything except `false` and `nil`).
+-- Throws an error when timeout exceeds.
+local function wait_for_condition(cond_desc, server, func, ...)
+    local deadline = clock.time() + WAIT_TIMEOUT
+    while true do
+        if func(...) then
+            return
+        end
+        if clock.time() > deadline then
+            error(('Timed out to wait for "%s" condition for server (alias: %s, workdir: %s, pid: %d) within %ds')
```

Comment:
Maybe a better order is this: alias, pid, workdir. The workdir can be a long string.

Reply:
Actually, it is not long because of `basename`.